library/public/types.h

```diff
@@ -82,6 +72,54 @@ struct mesh_t
    */
   F3D_EXPORT std::pair<bool, std::string> isValid() const;
 };
+
+/**
+ * Describe a 3D vector.
+ */
+struct vector3_t : std::array<double, 3>
```

Comment:
Let's not derive from std::array.
We cannot ensure that `_M_elems` is the variable name since it's implementation specific I think.
Just add `double Values[3]` as private membre.

Reply:
I could do this but it breaks code in a few places, since the std::array api is used through a vector3_t in many places.

I have added the changes for this, let me know if I should keep them or revert back.

Reply:
I prefer this version! Thanks for adding the required apis.